src/main/java/com/google/api/gax/core/ConnectionSettings.java

```diff
@@ -56,25 +56,19 @@
 @AutoValue
 public abstract class ConnectionSettings {
 
-  /*
-   * package-private so that the AutoValue derived class can access it
-   */
-  interface CredentialsProvider {
-    Credentials getCredentials() throws IOException;
-  }
-
   /**
-   * Gets the credentials which will be used to call the service. If the credentials
-   * have not been acquired yet, then they will be acquired when this function is called.
+   * Gets the credentials which will be used to call the service. If the credentials have not been
+   * acquired yet, then they will be acquired when this function is called.
    */
-  public Credentials getCredentials() throws IOException {
+  public Credentials getOrBuildCredentials() throws IOException {
```

Comment:
I think I would want the method name here and in CredentialsProvider to be consistent.